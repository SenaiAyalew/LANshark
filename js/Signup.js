/* eslint-disable import/no-cycle */
/* eslint-disable no-alert */
/* eslint-disable no-undef */


import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  TextInput,
  View,
} from 'react-native';
import axios from 'axios';
import Login from './Login';
import renderIf from './helpers/renderIf';

const styles = StyleSheet.create({
  login: {
    alignItems: 'center',
    alignSelf: 'center',
  },
  header: {
    textAlign: 'center',
    fontSize: 40,
    fontWeight: 'bold',
    color: '#fff',
    paddingBottom: 10,
    marginBottom: 40,
    borderBottomColor: '#199187',
    borderBottomWidth: 1,
  },
  textinput: {
    backgroundColor: 'white',
    fontSize: 22,
    alignSelf: 'stretch',
    height: 53,
    marginBottom: 20,
    borderBottomColor: '#f8f8f8',
    borderBottomWidth: 1,
  },
  signupbutton: {
    alignSelf: 'stretch',
    alignItems: 'center',
    padding: 20,
    backgroundColor: '#59cbbd',
    marginTop: 30,
  },
  logintext: {
    alignSelf: 'stretch',
    textAlign: 'center',
    marginTop: 30,
    color: '#fff',
    fontSize: 20,
    fontWeight: 'bold',
  },
  btntext: {
    color: '#fff',
    fontSize: 20,
    fontWeight: 'bold',
  },
});


export default class Signup extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: '',
      password: '',
      loginPage: false,
      signupPage: true,
    };

    this._loginPage = this._loginPage.bind(this);
    this._signup = this._signup.bind(this);
    this._signin = this._signin.bind(this);
  }


  _loginPage() {
    this.setState({
      loginPage: true,
      signupPage: false,
    });
  }

  _signup() {
    this.setState({
      signupPage: true,
      loginPage: false,
    });
  }

  _signin() {
    const deployedServer = 'http://ec2-54-152-18-28.compute-1.amazonaws.com/signup';

    axios({
      method: 'post',
      url: deployedServer,
      data: {
        email: this.state.email,
        password: this.state.password,
      },
    })
      .then((response) => {
        if (typeof response.data === 'object') {
          this.props.user.id = response.data.id;
          this.props._arView();
        } else if (response.data === 'User not created') {
          alert(`Sorry ${this.state.email}, something went wrong. Please try again.`);
        } else {
          alert(`Sorry ${this.state.email}, this email is Already registered. Try login.`);
        }
      })
      .catch((error) => {
        throw error;
      });
  }

  render() {
    return (
      <View style={styles.login}>
        {renderIf(this.state.signupPage && !this.state.loginPage,
          <View>
            <Text style={styles.header}>Welcome to AR History Tour</Text>

            <TextInput
              style={styles.textinput}
              placeholder="   Email"
              onChangeText={text => this.setState({ email: text })}
            />
            <TextInput style={styles.textinput} placeholder="   Password" secureTextEntry onChangeText={text => this.setState({ password: text })} />

            <TouchableOpacity
              style={styles.signupbutton}
              onPress={() => { this._signin(); }}
            >
              <Text style={styles.btntext}>Sign Up</Text>
            </TouchableOpacity>

            <Text style={styles.logintext} onPress={() => { this._loginPage(); }}>Login Here</Text>
          </View>)}
        {renderIf(!this.state.signupPage && this.state.loginPage,
          <View>
            <Login
              arView={this.props._arView}
              signup={this.props._signup}
              user={this.props.user}
            />
          </View>)}
      </View>
    );
  }
}
