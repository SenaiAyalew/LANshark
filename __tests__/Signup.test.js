import 'react-native';
import React from 'react';
import Signup from '../js/Signup';
import renderer from 'react-test-renderer';

test('Signup snapShot', () => {
  const snap = renderer.create(
    <Signup />
  ).toJSON();
expect(snap).toMatchSnapshot();
});

test('Email in state should be an empty string', () => {
  const SignComp = renderer.create(
    <Signup />
  ).getInstance();

expect(SignComp.state.email).toBe('');
});

test('Password in state should be an empty string', () => {
  const SignComp = renderer.create(
    <Signup />
  ).getInstance();

expect(SignComp.state.password).toBeFalsy();
});

test('loginPage in state should be false', () => {
  const SignComp = renderer.create(
    <Signup />
  ).getInstance();

expect(SignComp.state.loginPage).toBeFalsy();
});

test('signupPage in state should be true', () => {
  const SignComp = renderer.create(
    <Signup />
  ).getInstance();

expect(SignComp.state.signupPage).toBeTruthy();
});

test('loginPage in state should be true when _loginPage is called', () => {
  const SignComp = renderer.create(
    <Signup />
  ).getInstance();
SignComp._loginPage();
expect(SignComp.state.loginPage).toBeTruthy();
});

test('loginPage in state should be true when _loginPage is called', () => {
  const SignComp = renderer.create(
    <Signup />
  ).getInstance();
SignComp._loginPage();
expect(SignComp.state.signupPage).toBeFalsy();
});

test('signupPage in state should be true when _signupPage is called', () => {
  const SignComp = renderer.create(
    <Signup />
  ).getInstance();
SignComp._signup();
expect(SignComp.state.signupPage).toBeTruthy();
});

test('loginPage in state should be false when _signupPage is called', () => {
  const SignComp = renderer.create(
    <Signup />
  ).getInstance();
SignComp._signup();
expect(SignComp.state.loginPage).toBeFalsy();
});

// test('_signup() should return an object', (done) => {
//   const SignComp = renderer.create(
//     <Signup />
//   ).getInstance();
//   expect.assertions(1);
//   return SignComp._signin().then((response) => {
//     expect(typeof response.data).toBe('object');
//     done();
//   });
// });

test('signupPage in state should be true when _signupPage is called', () => {
  const SignComp = renderer.create(
    <Signup />
  ).getInstance();
expect(typeof SignComp.styles).toBeTruthy();
});
