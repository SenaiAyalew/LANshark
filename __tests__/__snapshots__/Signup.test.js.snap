// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login snapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "borderBottomColor": "#199187",
          "borderBottomWidth": 1,
          "color": "#fff",
          "fontSize": 40,
          "fontWeight": "bold",
          "marginBottom": 40,
          "paddingBottom": 10,
          "textAlign": "center",
        }
      }
    >
      Welcome to AR History Tour
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="   Email"
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "borderBottomColor": "#f8f8f8",
          "borderBottomWidth": 1,
          "fontSize": 22,
          "height": 53,
          "marginBottom": 20,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="   Password"
      secureTextEntry={true}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "borderBottomColor": "#f8f8f8",
          "borderBottomWidth": 1,
          "fontSize": 22,
          "height": 53,
          "marginBottom": 20,
        }
      }
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#59cbbd",
          "marginTop": 30,
          "opacity": 1,
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Sign Up
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginTop": 30,
          "textAlign": "center",
        }
      }
    >
      Login Here
    </Text>
  </View>
</View>
`;

exports[`Signup snapShot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "borderBottomColor": "#199187",
          "borderBottomWidth": 1,
          "color": "#fff",
          "fontSize": 40,
          "fontWeight": "bold",
          "marginBottom": 40,
          "paddingBottom": 10,
          "textAlign": "center",
        }
      }
    >
      Welcome to AR History Tour
    </Text>
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="   Email"
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "borderBottomColor": "#f8f8f8",
          "borderBottomWidth": 1,
          "fontSize": 22,
          "height": 53,
          "marginBottom": 20,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="   Password"
      secureTextEntry={true}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "white",
          "borderBottomColor": "#f8f8f8",
          "borderBottomWidth": 1,
          "fontSize": 22,
          "height": 53,
          "marginBottom": 20,
        }
      }
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "backgroundColor": "#59cbbd",
          "marginTop": 30,
          "opacity": 1,
          "padding": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#fff",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }
      >
        Sign Up
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "color": "#fff",
          "fontSize": 20,
          "fontWeight": "bold",
          "marginTop": 30,
          "textAlign": "center",
        }
      }
    >
      Login Here
    </Text>
  </View>
</View>
`;
